tslearn/bases.py

```diff
@@ -0,0 +1,236 @@
+
+        Parameters
+        ----------
+        cls : instance of model that inherits from `BaseModelPackage`
+            a model instance
+
+        model : dict
+            Model dict containing hyper-parameters and model-parameters
+
+        Returns
+        -------
+        model: instance of model that inherits from `BaseModelPackage`
+            instance of the model class with hyper-parameters and
+            model parameters set from the passed model dict
+        """
+
+        model_params = model.pop('model_params')
+        hyper_params = model.pop('hyper_params')  # hyper-params
+
+        # instantiate with hyper-parameters
+        inst = cls(**hyper_params)
+
+        # set all model params
+        for p in model_params.keys():
+            setattr(inst, p, model_params[p])
+
+        return inst
+
+    def to_hdf5(self, path):
+        """
+        Save this model as an HDF5 file.
+
+        Parameters
+        ----------
+        path : str
+            Full file path. File must not already exist.
+
+        Returns
+        -------
+        None
+
+        Raises
+        ------
+        FileExistsError
+            If a file with the same path already exists.
+        """
+
+        d = self._to_dict(output='hdf5')
+        hdftools.save_dict(d, path, 'data')
+
+    @classmethod
+    def from_hdf5(cls, path):
+        """
+        Load from an HDF5 file
+
+        Parameters
+        ----------
+        path : str
+            Full path to file.
+
+        Returns
+        -------
+        Model instance
+        """
+
+        model = hdftools.load_dict(path, 'data')
+
+        for k in model['hyper_params'].keys():
+            if model['hyper_params'][k] == 'None':
+                model['hyper_params'][k] = None
+
+        return BaseModelPackage._organize_model(cls, model)
+
+    def to_json(self, path):
+        """
+        Save as a JSON file.
+
+        Parameters
+        ----------
+        path : str
+            Full file path.
+
+        Returns
+        -------
+        None
+        """
+
+        d = self._to_dict(output='json')
+        json.dump(d, open(path, 'w'))
+
+    @classmethod
+    def from_json(cls, path):
+        """
+        Load from a json file.
+
+        Parameters
+        ----------
+        path : str
+            Full path to file.
+
+        Returns
+        -------
+        Model instance
+        """
+
+        model = json.load(open(path, 'r'))
+
+        # Convert the lists back to arrays
+        for k in model['model_params'].keys():
+            param = model['model_params'][k]
+            if type(param) is list:
+                model['model_params'][k] = np.array(param)
+
+        return BaseModelPackage._organize_model(cls, model)
+
+    def to_pickle(self, path):
+        """
+        Save as a pickle. Not recommended for interoperability.
+
+        Parameters
+        ----------
+        path : str
+            Full file path.
+
+        Returns
```

Comment:
For all `to_*` methods, do not have a `Returns` section since they do not return anything.